test/test_ops.py

```diff
@@ -339,11 +350,45 @@ def test_out(self, device, dtype, op):
         sample = samples[0]
         # call it normally to get the expected result
         expected = op(*sample.input, *sample.args, **sample.kwargs)
-        # call it with out=... and check we get the expected result
-        out_kwargs = sample.kwargs.copy()
-        out_kwargs['out'] = out = torch.empty_like(expected)
-        op(*sample.input, *sample.args, **out_kwargs)
-        self.assertEqual(expected, out)
+
+        def _test(tested_op):
+            # call it with out=... and check we get the expected result
+            out_kwargs = sample.kwargs.copy()
+            out_kwargs['out'] = out = torch.empty_like(expected)
+            tested_op(*sample.input, *sample.args, **out_kwargs)
+            self.assertEqual(expected, out)
+
+        _test(op)
+        for a_op in op.aliases:
+            _test(a_op)
+
+    @ops([op for op in op_db if op.aliases])
+    def test_aliases_jit_name_remapping(self, device, dtype, op):
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
```

Comment:
This is on the right path. Here's my attempt at annotating a modified version of the current test:

```
#  decorator goes here
def test_jit_alias_remapping(self, device, dtype, op):
  tensor = torch.tensor
  # acquire all op variants and note which are inplace
  for each alias op variant:
    # determine if it's inplace
    # get the string name of the variant (e.g. "absolute", "absolute_")
    # get the corresponding "original name" (e.g. "abs", "abs_")
    # acquire the args from the first sample input

    # Checks that scripting converts aliases
    # NOTE: the code to test scripting must be generated since
    #   scripting does not support splatting args or directly
    #   calling torch.Tensor methods. The following
    #   splats args after the first tensor by inlining them as constants.
    if is_inplace:
      fn_template = '''
        def _fn(t):
          return t.{alias_name}({args})
      '''
      arg_string = ', '.join((str(arg) for arg in info.get_args(device)))  # modify this to use acquired args
      script = fn_template.format(alias_name=info.alias_name, args=arg_string)
    else:
      is_input_tensor_list = isinstance(info.get_input(device), Sequence)
      # For sequence of Tensors, annotate the type to be List[Tensor]
      if is_input_tensor_list:
        fn_template = '''
          def _fn(t: List[Tensor]):
            return op(t{args})
        '''
      else:
        fn_template = '''
          def _fn(t):
           return op(t{args})
          '''
        arg_string = ", " + ', '.join((str(arg) for arg in info.get_args(device)))  # modify this to use args
        script = fn_template.format(args=arg_string)

    # Compiles script
    scripted = torch.jit.CompilationUnit(script)._fn 

    # Acquires and checks the graph remaps the alias
    inp = info.get_input(device)  # this can be acquired from sample inputs now
    # input will have to be cloned for inplace ops. Maybe cloning should be added to SampleInputs since it 
    #  comes up so much? The clone only needs to copy the input tensor.
    scripted(clone_inp(inp))  
    graph = scripted.graph_for(clone_inp(inp))
    FileCheck().check(info.original_name).check_not(info.alias_name).run(graph)
    # Note at this point we're done with SCRIPTING, now we want to check TRACING

    # Checks that tracing converts aliases
    # NOTE: tracing has no problem splatting args
    args = info.get_args(device)  # this will need to be updated to use the args acquired earlier

    def _fn(t, info=info, args=args):
      return info.alias_op(t, *args)

    # the cloned inputs here can be handled like they are above
    traced = torch.jit.trace(_fn, (clone_inp(inp),))
    traced(clone_inp(inp))
    graph = traced.graph_for(clone_inp(inp))
    FileCheck().check(info.original_name).check_not(info.alias_name).run(graph)
```

Reply:
Oh, I see. Thanks! Sorry, I didn't understand that we'd like to port completely the code from test_op_aliases.py.